Java/libraries/recognizers-text-datatypes-timex-expression/src/main/java/com/microsoft/recognizers/datatypes/timex/expression/TimexResolver.java

```diff
@@ -249,80 +249,78 @@ private static String nextDateValue(TimexProperty timex, Calendar date) {
     }
 
     private static Pair<String, String> yearWeekDateRange(Integer year, Integer weekOfYear, Boolean isWeekend) {
-        Calendar firstMondayInWeek = firstDateOfWeek(year, weekOfYear, Locale.ROOT);
+        LocalDateTime firstMondayInWeek = firstDateOfWeek(year, weekOfYear, Locale.ROOT);
 
-        Calendar start = (isWeekend == null || isWeekend == false) ? firstMondayInWeek
+        LocalDateTime start = (isWeekend == null || isWeekend == false) ? firstMondayInWeek
                 : TimexDateHelpers.dateOfNextDay(DayOfWeek.SATURDAY, firstMondayInWeek);
-        Calendar end = firstMondayInWeek;
-        end.add(Calendar.DATE, 7);
+        LocalDateTime end = firstMondayInWeek;
+        LocalDateTime end2 = end.plusDays(7);
 
         return Pair.of(TimexValue.dateValue(new TimexProperty() {
             {
-                setYear(start.get(Calendar.YEAR));
-                setMonth(start.get(Calendar.MONTH));
-                setDayOfMonth(start.get(Calendar.DATE));
+                setYear(start.getYear());
+                setMonth(start.getMonthValue());
+                setDayOfMonth(start.getDayOfMonth());
             }
         }), TimexValue.dateValue(new TimexProperty() {
             {
-                setYear(end.get(Calendar.YEAR));
-                setMonth(end.get(Calendar.MONTH));
-                setDayOfMonth(end.get(Calendar.DATE));
+                setYear(end2.getYear());
+                setMonth(end2.getMonthValue());
+                setDayOfMonth(end2.getDayOfMonth());
             }
         }));
     }
 
     // this is based on
     // https://stackoverflow.com/questions/19901666/get-date-of-first-and-last-day-of-week-knowing-week-number/34727270
-    private static Calendar firstDateOfWeek(Integer year, Integer weekOfYear, Locale cultureInfo) {
+    private static LocalDateTime firstDateOfWeek(Integer year, Integer weekOfYear, Locale cultureInfo) {
         // ISO uses FirstFourDayWeek, and Monday as first day of week, according to
         // https://en.wikipedia.org/wiki/ISO_8601
-        Calendar jan1 = Calendar.getInstance();
-        jan1.set(year, 1, 1);
-        Integer daysOffset = DayOfWeek.MONDAY.getValue() - jan1.get(Calendar.DAY_OF_WEEK);
-        Calendar firstWeekDay = jan1;
-        firstWeekDay.add(Calendar.DATE, daysOffset);
+        LocalDateTime jan1 = LocalDateTime.of(year, 1, 1, 0, 0);
+        Integer daysOffset = DayOfWeek.MONDAY.getValue() - jan1.getDayOfWeek().getValue();
+        LocalDateTime firstWeekDay = jan1;
+        firstWeekDay = firstWeekDay.plusDays(daysOffset);
 
-        Integer firstWeek = jan1.get(Calendar.WEEK_OF_YEAR);
+        Integer firstWeek = jan1.getDayOfYear();
```

Comment:
Is this correct? the removed line was getting the week and in the line you added you are getting the day of the year

Reply:
You are right!

Reply:
Applied!